lib/APINotes/APINotesYAMLCompiler.cpp

```diff
@@ -164,7 +164,15 @@ namespace {
     AvailabilityItem() : Mode(APIAvailability::Available), Msg("") {}
   };
 
-  static llvm::Optional<NullabilityKind> AbsentNullability = llvm::None;
+  // There is a bug in MSVC that causes an internal compiler error in compilation.
+  // See: https://connect.microsoft.com/VisualStudio/feedback/details/3111599/
+  // This is a workaround that issue until the problem gets fixed.
+#if defined(_MSC_VER)
+  static llvm::Optional<NullabilityKind> AbsentNullability = llvm::NoneType::None;
```

Comment:
This worked, thanks Alex.

Curious, the following: `llvm::Optional< NullabilityKind > AbsentNullability = {};` *didn't* work: I presumed you made a typo msising the `=`. Clearly you didn't :D. Do you know why?